app/components/markdown/markdown_link/markdown_link.js

```diff
@@ -55,7 +57,16 @@ export default class MarkdownLink extends PureComponent {
         const match = matchDeepLink(url, serverUrl, siteURL);
         if (match) {
             if (match.type === DeepLinkTypes.CHANNEL) {
-                this.props.actions.handleSelectChannelByName(match.channelName, match.teamName);
+                const response = await this.props.actions.handleSelectChannelByName(match.channelName, match.teamName);
+
+                if (response && response.error) {
+                    const linkFailedMessage = {
+                        id: t('permalink.error.access'),
+                        defaultMessage: 'Permalink belongs to a deleted message or to a channel to which you do not have access.',
```

Comment:
@adamjclarkson maybe change this to something like

```suggestion
                        defaultMessage: 'You are not a member of this private channel.',
```

or something along those lines, thoughts?

Reply:
Yep. I like it. More succinct. Thanks Elias.